.circleci/config.yml

```diff
@@ -62,47 +36,34 @@ aliases:
       paths:
         - ~/.cache/yarn
 
-  - &save_node_modules
-    save_cache:
-      name: Save node_modules cache
-      # Cache only for the current revision to prevent cache injections from
-      # malicious PRs.
-      key: v1-node_modules-{{ arch }}-{{ .Revision }}
-      paths:
-        - node_modules
-        - packages/eslint-plugin-react-hooks/node_modules
-        - packages/react-art/node_modules
-        - packages/react-client/node_modules
-        - packages/react-devtools-core/node_modules
-        - packages/react-devtools-extensions/node_modules
-        - packages/react-devtools-inline/node_modules
-        - packages/react-devtools-shared/node_modules
-        - packages/react-devtools-shell/node_modules
-        - packages/react-devtools-timeline/node_modules
-        - packages/react-devtools/node_modules
-        - packages/react-dom/node_modules
-        - packages/react-interactions/node_modules
-        - packages/react-native-renderer/node_modules
-        - packages/react-reconciler/node_modules
-        - packages/react-server-dom-relay/node_modules
-        - packages/react-server-dom-webpack/node_modules
-        - packages/react-server-native-relay/node_modules
-        - packages/react-server/node_modules
-        - packages/react-test-renderer/node_modules
-        - packages/react/node_modules
-        - packages/scheduler/node_modules
-
-  - &restore_node_modules
-    restore_cache:
-      name: Restore node_modules cache
-      keys:
-        - v1-node_modules-{{ arch }}-{{ .Revision }}
-
   - &TEST_PARALLELISM 20
 
   - &attach_workspace
     at: build
 
+commands:
+  setup_node_modules:
+    description: "Restore node_modules"
+    steps:
+      - restore_cache:
+          name: Restore yarn cache
+          keys:
+            - v1-yarn_cache-{{ arch }}-{{ checksum "yarn.lock" }}
+            - v1-yarn_cache-{{ arch }}-
+            - v1-yarn_cache-
+      - run:
+          name: Install dependencies
+          command: |
+            yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
+            if [ $? -ne 0 ]; then
+              yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
```

Comment:
Retry in case of network related failure?

Reply:
Yes, I think in the past we've seen intermittent flakiness here. Doesn't seem bad to have 1 retry.